src/app/components/userprofile/userprofile.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit } from '@angular/core';
+import { LoggedUser } from "../../model/logged-user";
+import { AuthService } from '../../services';
+import { environment } from 'src/environments/environment';
+import { AngularFireStorage } from '@angular/fire/storage';
+import { HttpHeaders, HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { User } from 'src/app/model';
+
+@Component({
+  selector: 'app-userprofile',
+  templateUrl: './userprofile.component.html',
+  styleUrls: ['./userprofile.component.css']
+})
+export class UserprofileComponent implements OnInit {
+  loggedUser: LoggedUser;
+  isUser = false;
+  imageProfile: string;
+  user: User;
+  newUser: User;
+  name: string;
+  preferredEmail: string;
+  phoneNumber: string;
+  pictureUrl: string;
+
+  constructor(private authService: AuthService, private httpClient: HttpClient, private storage: AngularFireStorage) {
+    if (this.authService.isAuthenticated()) {
+      this.loggedUser = this.authService.getLoggedUser();
+      if (!this.isUser) {
+        this.getMyInformation().subscribe();
+      }
+    }
+  }
+
+  ngOnInit() {
+  }
+  protected getMyInformation(): Observable<User> {
+    const urlEndpoint = `${environment.backendUrl}/api/user/me`;
+    const httpHeaders = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Authorization': `Bearer ${this.loggedUser.access_token}`,
+    });
+    return this.httpClient.get<User>(urlEndpoint, { headers: httpHeaders })
```

Comment:
A component shouldn't interact with backend .. this is a job for a service layer.